zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/GlobalControllerExceptionHandler.java

```diff
@@ -42,6 +43,11 @@ protected ProblemDetail createProblemDetail(
       // with "Required request body is missing"
       // for proper exception tracing
       detail = REQUEST_BODY_MISSING_EXCEPTION_MESSAGE;
+    } else if (isJsonParseError(ex)) {
+      // Replace detail "Failed to read request"
+      // with "JSON parse error"
+      // for proper exception tracing
+      detail = "Unable to parse JSON request body";
```

Comment:
❓ What's the full original error message? Could we maybe detect that we have an enum problem so we can hint at that? I guess the more detailed the information, the easier users can fix their requests. Just trying to explore what is possible here.

Reply:
The error that we get is
```
JSON parse error: Cannot construct instance of `io.camunda.zeebe.gateway.protocol.rest.Permission`, problem: Unexpected value 'UNKNOWN'
```
with a root cause of
```
java.lang.IllegalArgumentException: Unexpected value 'UNKNOWN'
```
I'm not sure we can confidently make assumptions here, but what if we took the original message and returned an error of something like 
```
Unexpected value 'UNKNOWN' for class 'io.camunda.zeebe.gateway.protocol.rest.Permission'
```

Reply:
👍 OK, I see, thanks for those details!
👍 I like the proposal in general.
🤔 Exposing the full class name including the package might not be helpful to users, though. In the REST API explorers in the public docs and the OpenAPI description, they will only see the type `Permission`, I guess? Will they even see this as a dedicated type or just an array of String options for that parameter, e.g. when using SwaggerUI to look at the OpenAPI description?

But it might become a bit error-prone if we do something like the following
```
Unexpected value 'UNKNOWN' for type 'Permission'
```

Reply:
Looking at the openapi spec generated the user is aware of the type 
<img width="503" alt="Screenshot 2024-07-03 at 08 30 34" src="https://github.com/camunda/camunda/assets/10366708/1e576c4d-26ec-499f-8fa9-81563068f01e">
so we can try to make this a little more intelligent and pull parts of the information out. I'll have a closer look :)

Reply:
Nice, I see. Yeah, that would be super helpful, I think. Especially, if this is generalized in the global mapper.